src/iframe-helper.js

```diff
@@ -380,3 +380,61 @@ function parseIfNeeded(data) {
   }
   return data;
 }
+
+
+
+/**
+ * Manages a postMessage API for an iframe with a subscription message and
+ * a way to broadcast messages to all subscribed windows, which
+ * in turn must all be descendants of the contentWindow of the iframe.
+ */
+export class SubscriptionApi {
```

Comment:
> As you can see, AMP cares about performance so much. In many places, we are trying hard to inline function calls to have some performance gain.

Only code with loops. This will only be done once per iframe (usually), so micro-opts won't affect real performance.

The code duplication that this fixes is a much bigger benefit, IMO.